pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -330,10 +331,10 @@ func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		lbIP = aviSECIDR
 	} else if lbProvider == vpc.NSXLB {
-		if vpcConnectivityProfile.ServiceGateway != nil && len(vpcConnectivityProfile.ServiceGateway.EdgeClusterPaths) != 0 {
+		if gatewayConnectionReady {
 			// CTGW is used for NSX LB
 			nsxLBSNATIP, err = r.Service.GetNSXLBSNATIP(*createdVpc, "gateway-interface")
-		} else if vpcConnectivityProfile.ServiceGateway != nil && len(vpcConnectivityProfile.ServiceGateway.ServiceClusterPaths) != 0 {
+		} else if serviceClusterReady {
```

Comment:
could we also add or change unit tests for this logic change?

Reply:
We already have UT for autocreated vpc getting lbsnatip. Added a unit test for precreated vpc with nsxlb, thanks!